pkg/retry/retry.go

```diff
@@ -38,17 +47,26 @@ func WithMaxAttempts(a int) Option {
 func Do(fn func() error, opts ...Option) error {
 	cfg := &config{
 		interval: 30 * time.Second,
+		backoff: false
 	}
 	for _, opt := range opts {
 		opt(cfg)
 	}
 
 	attempts := 0
 	ticker := time.NewTicker(cfg.interval)
+	tickerMax := time.Duration(5*cfg.interval) * cfg.interval
 	defer ticker.Stop()
 
 	for {
 		attempts++
+		if cfg.backoff == true {
+			if (cfg.interval * time.Duration(attempts) <= tickerMax {
+				ticker = time.NewTicker(cfg.interval * time.Duration(attempts))
+			} else {
+				ticker = tickerMax
```

Comment:
This is leaking tickers, `ticker.Stop` needs to be called after we are done using a ticker created with `time.NewTicker` (see https://pkg.go.dev/time#NewTicker).